client-hc/src/main/java/com/graphhopper/api/MatrixResponse.java

```diff
@@ -188,6 +230,39 @@ public String toString() {
             addInfo += ", distances: " + distances.length + "x" + distances[0].length;
         }
 
-        return "[" + addInfo + "] errors:" + errors.toString();
+        String result = "[" + addInfo + "] errors:" + errors.toString();
+        if (!disconnectedPoints.isEmpty()) {
+            result += ", disconnectedPoints: " + disconnectedPoints.size();
+        }
+        if (!invalidFromPoints.isEmpty()) {
+            result += ", invalidFromPoints: " + invalidFromPoints.size();
+        }
+        if (!invalidToPoints.isEmpty()) {
+            result += ", invalidToPoints: " + invalidToPoints.size();
+        }
+        return result;
+    }
+
+    public static class PointPair {
+        private final int sourceIndex;
+        private final int targetIndex;
```

Comment:
(`public final` would be also fine for those structs allowing as to avoid the getter stuff)

Reply:
ok changed it.